modules/primo/includes/primo.search.inc

```diff
@@ -178,7 +178,23 @@ function primo_search_search(TingSearchRequest $ting_query) {
 
   // Handle raw query.
   if (!empty($raw_query = $ting_query->getRawQuery())) {
-    $queries['query'][] = htmlspecialchars($raw_query);
+    // The query starts with "[", so we detect if we are handling raw
+    // material ID's or a free text search.
+    if (strpos($raw_query, '[') !== FALSE) {
+      $raw_queries = explode(',', trim($raw_query, '[]'));
+      $separate_queries = [];
+      foreach ($raw_queries as $key => $raw_query) {
+        $institution_code = variable_get('primo_institution_code');
+        $starts_with_institution_code = preg_match("/^$institution_code/", $raw_query);
+        $rid_query = $key === 0 ? "rid,contains,$raw_query" : $raw_query;
```

Comment:
All the rids has to go into a single query - someting ala 'rid,contains,' . implode(' OR ', $raw_queries) at which point you're bascially doing what we should already be doing for $query->getMaterialFilter(). Titles should then go into a seperate 'swtitle, begins_with,' . implode(' OR ', $raw_queries)